src/emulator.ts

```diff
@@ -94,6 +94,9 @@ export const createEmulator = async (name: string, options: EmulatorOptions) =>
     for (const dir of ['platforms', 'platform-tools'])
         if (!(await fs.pathExists(join(androidHome, dir)))) await fs.ensureDir(join(androidHome, dir));
 
+    // The `emulator` package is required by avdmanager, so we need to check for it
+    if (!(await isInstalled('emulator'))) await installPackages('emulator');
```

Comment:
While it doesn't take forever, the `sdkmanager` command here isn't particularly fast, either. I don't love having this overhead every time you create an emulator.

Any reason you're not just checking for the existence of the directory as we're doing in all other cases? I verified that `sdkmanager --list_installed` doesn't do anything more fancy, either.[^n]

[^n]: By that I mean it doesn't check whether an existing installation is corrupt or anything like that. It just checks for the presence of `$ANDROID_HOME/emulator/package.xml` (and parses that to show the version information, but we don't need that…).

The changes in `listPackages` are useful regardless.